manifests/passenger.pp

```diff
@@ -3,39 +3,58 @@
 # Apache resources specific for passenger
 #
 class apache::passenger {
-
   include apache
 
   case $::operatingsystem {
-    ubuntu,debian,mint: {
-      package { 'libapache2-mod-passenger':
-        ensure => present;
-      }
+    ubuntu, debian, mint : {
+      package { 'libapache2-mod-passenger': ensure => present; }
 
       exec { 'enable-passenger':
         command => '/usr/sbin/a2enmod passenger',
         creates => '/etc/apache2/mods-enabled/passenger.load',
         notify  => Service['apache'],
-        require => [
-          Package['apache'],
-          Package['libapache2-mod-passenger']
-        ],
+        require => [Package['apache'], Package['libapache2-mod-passenger']],
       }
     }
 
-    centos,redhat,scientific,fedora: {
+    centos, redhat, scientific, fedora : {
       $osver = split($::operatingsystemrelease, '[.]')
 
       case $osver[0] {
-        5: { require yum::repo::passenger }
-        default: { }
-      }
-      package { 'mod_passenger':
-        ensure => present;
+        5       : {
+          require yum::repo::passenger
+
+          package { 'mod_passenger': ensure => present; }
+        }
+        7       : {
+          $version = '5.0.15'
+          $build_tools = ["gcc-c++", "libcurl-devel", "openssl-devel", "zlib-devel", "httpd-devel", "apr-devel", 'apr-util-devel']
+
+          if !defined(Class['::ruby::dev']) {
+            class { '::ruby::dev': }
+          }
+
+          package { $build_tools: } ->
```

Comment:
We can't place there common packages here, they might be already declared in existing setups, a mitigating approach would be to use ensure_resources, but still I 'd move their installation in a dedicated subclass which might be disabled via parameters. Also, do we really need to compile the whole stuff? No packages available from some yum repo?